reactive/kotlinx-coroutines-rx2/test/SchedulerTest.kt

```diff
@@ -31,4 +38,252 @@ class SchedulerTest : TestBase() {
+                val t1 = Thread.currentThread()
+                assertSame(t1, mainThread)
+                expect(2)
+                it.resume(Unit)
+            }, -1, TimeUnit.MILLISECONDS)
+        }
+        yield()
+        finish(3)
+    }
+
+    @Test
+    fun testDisposeDuringDelay(): Unit = runTest {
+        expect(1)
+        val scheduler = (currentDispatcher() as CoroutineDispatcher).asScheduler()
+        val delayMillis = 300L
+        val disposable = scheduler.scheduleDirect({
+            expectUnreached()
+        }, delayMillis, TimeUnit.MILLISECONDS)
+        delay(100)
+        expect(2)
+        disposable.dispose()
+        delay(300)
+        yield()
+        finish(3)
+    }
+
+    @Test
+    fun testImmediateDispose(): Unit = runTest {
+        expect(1)
+        val scheduler = (currentDispatcher() as CoroutineDispatcher).asScheduler()
+        val disposable = scheduler.scheduleDirect {
+            expectUnreached()
+        }
+        disposable.dispose()
+        yield()
+        finish(2)
+    }
+
+    @Test
+    fun testAsSchedulerWorksWithSchedulerCoroutineDispatcher(): Unit = runTest {
+        expect(1)
+
+        val dispatcher = Schedulers.io().asCoroutineDispatcher()
+        val scheduler = dispatcher.asScheduler()
+        suspendCancellableCoroutine<Unit> {
+            scheduler.scheduleDirect {
+                expect(2)
+                it.resume(Unit)
+            }
+        }
+        finish(3)
+    }
+
+    @Test
+    fun tesConvertDispatcherToOriginalScheduler(): Unit = runTest {
+        expect(1)
+
+        val originalScheduler = Schedulers.io()
+        val dispatcher = originalScheduler.asCoroutineDispatcher()
+        val scheduler = dispatcher.asScheduler()
+        assertEquals(originalScheduler, scheduler)
+
+        finish(2)
+    }
+
+    @Test
+    fun tesConvertSchedulerToOriginalDispatcher(): Unit = runTest {
+        expect(1)
+
+        val originalDispatcher = currentDispatcher() as CoroutineDispatcher
+        val scheduler = originalDispatcher.asScheduler()
+        val dispatcher = scheduler.asCoroutineDispatcher()
+        assertEquals(originalDispatcher, dispatcher)
+
+        finish(2)
+    }
+
+    @Test
+    fun testExpectRxPluginsCall(): Unit = runTest {
+        expect(1)
+
+        fun setScheduler(expectCountOnRun: Int) {
+            RxJavaPlugins.setScheduleHandler(Function {
+                Runnable {
+                    expect(expectCountOnRun)
+                    it.run()
+                }
+            })
+        }
+
+        val dispatcher = currentDispatcher() as CoroutineDispatcher
+        val scheduler = dispatcher.asScheduler()
+
+        setScheduler(2)
+
+        suspendCancellableCoroutine<Unit> {
+            scheduler.scheduleDirect {
+                expect(3)
+                it.resume(Unit)
+            }
+        }
+
+        setScheduler(4)
+        suspendCancellableCoroutine<Unit> {
+            scheduler.scheduleDirect({
```

Comment:
This is testing two different functions, it should be two separate tests. This makes it more clear what's actually broken if it fails, and also ensures that you're not accidentally relying on (or being broken by) shared state between the two function calls.